### PR TITLE
Add "Status" field to gql Enrollment

### PR DESCRIPTION
**Pull request by gigxz**
...that incorporates the overall household status. To be used in some table views.

### CODE REVIEW

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    value 'ANY_ENTRY_INCOMPLETE', 'Household Entry Incomplete'
+    value 'ACTIVE', 'Active'
+    value 'EXITED', 'Exited'
+    value 'OWN_EXIT_INCOMPLETE', 'Exit Incomplete'
+    value 'ANY_EXIT_INCOMPLETE', 'Household Exit Incomplete'
+
+    def self.from_enrollment(enrollment, user:)
+      return 'OWN_ENTRY_INCOMPLETE' if enrollment.in_progress?
+
+      return 'OWN_EXIT_INCOMPLETE' if enrollment.exit_date.nil? && enrollment.exit_assessment&.present?
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: Can you have an exit_assessment with no ExitDate (maybe that's the WIP version?)

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      household_members = Hmis::Hud::Enrollment.viewable_by(user).where(household_id: enrollment.household_id)
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: Hmm, should we include project_id in this too?  I _think_ household_id should be unique across a data source, but I'm always a bit nervous about that assumption.

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      return 'ANY_ENTRY_INCOMPLETE' if household_members.count > 1 && household_members.in_progress.exists?
+      return 'ANY_EXIT_INCOMPLETE' if household_members.count > 1 && household_members.find(&:exit_assessment).present?
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: This is going to say the exit is incomplete if one person has an exit_assessment right?  I'm unclear what the find on the household_members does.
Maybe an exit_assessment indicates WIP?

**Review by eanders — APPROVED** [review]
This looks good, I think I'm confused about some of the details about what makes an assessment/record WIP, incomplete, or active/exited. Is it worth moving those calculations into the record classes?  So you could ask `enrollment.exit_complete?` or something like that?

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: Yes. If you have a WIP Exit Assessment, then you don't have an ExitDate.

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: Oh yikes, really? Last time household ID came up [in a PR](https://github.com/greenriver/hmis-warehouse/pull/2838#discussion_r1115860213), you seemed confident that it was unique across a data source. I made a change to ensure that we always use `viewable_by` when looking up a household ID, to ensure that uniqueness. I'll continue the conversation in Slack

File: drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb (hunk 3 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/enums/enrollment_status.rb]: Oh good catch. I think this is wrong. It needs a `where(ExitDate: nil)`.

"ANY_EXIT_INCOMPLETE" means that there is _another_ household member (not this one) who has an "incomplete exit." An enrollment has a "incomplete exit" if it's unexited but an Exit assessment has been started.

**Comment by gigxz** [comment]
@eanders yeah, I did that with https://github.com/greenriver/hmis-warehouse/pull/2915/commits/45a8cb3dc67bca9e8e57f515b19576152ff592ba. I don't think the querying is efficient at all, but maybe its ok since its just a household.

**Pull request merged**